invisible_cities/io/hist_io_test.py

```diff
@@ -0,0 +1,105 @@
+import os
+import string
+
+import numpy  as np
+import tables as tb
+
+from pytest import raises
+
+from hypothesis             import assume
+from hypothesis             import given
+from hypothesis             import settings
+from hypothesis             import HealthCheck
+from hypothesis.extra.numpy import arrays
+from hypothesis.strategies  import composite
+from hypothesis.strategies  import integers
+from hypothesis.strategies  import floats
+from hypothesis.strategies  import text
+from hypothesis.strategies  import lists
+from hypothesis.strategies  import sampled_from
+from hypothesis.strategies  import one_of
+
+from .. evm.histos          import HistoManager, Histogram
+from .. io .hist_io         import save_histomanager_to_file
+from .. io .hist_io         import get_histograms_from_file
+from .. evm.histos_test     import histograms_lists
+from .. evm.histos_test     import assert_histogram_equality
+
+
+characters = tuple(string.ascii_letters + string.digits + "-")
+
+@settings(suppress_health_check=(HealthCheck.too_slow,))
+@given   (histograms_lists(), text(characters, min_size=1))
+def test_save_histomanager_to_file_write_mode(output_tmpdir, histogram_list, group):
+    args, list_of_histograms = histogram_list
+    histogram_manager        = HistoManager(list_of_histograms)
+
+    file_out = os.path.join  (output_tmpdir    , 'test_save_histogram_manager.h5')
+    save_histomanager_to_file(histogram_manager, file_out, group=group)
+
+    with tb.open_file(file_out, "r") as h5in:
+        file_group = getattr(h5in.root, group)
+        for histogram in list_of_histograms:
+            histoname    = histogram.title
+            saved_labels = [str(label)[2:-1].replace('\\\\', '\\') for label in getattr(file_group, histoname + "_labels")[:]]
+
+            assert histoname in file_group
+            assert np.all     (a == b for a, b in zip(histogram.bins, getattr(file_group, histoname + "_bins")[:]))
```

Comment:
You need to ensure the length of  `histogram.bins` and `getattr(file_group, histoname + "_bins")[:]`matches. `zip` truncates the output to the shorter of the inputs.

Reply:
I guess we have discussed it a few times, but why is it not enough `np.allclose(histogram.bins, getattr(file_group, histoname + "_bins")[:]`?

Reply:
For 2D plots the bins don't have de same dimension so the bins are a array of dtype=object which I think fails when using the allclose.